app/src/test/java/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModelTest.kt

```diff
@@ -620,4 +626,102 @@ class ConnectionsListViewModelTest : ViewModelTest() {
             )
         )
     }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase1() {
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_NEGATIVE,
+            actionResId = R.string.actions_proceed
+        )
+
+        assertNull(viewModel.onAskPermissionsEvent.value)
+
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_POSITIVE,
+            actionResId = R.string.actions_proceed
+        )
+
+        assertNotNull(viewModel.onAskPermissionsEvent.value)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase2() {
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_NEGATIVE,
+            actionResId = R.string.actions_go_to_settings
+        )
+
+        assertNull(viewModel.onGoToSettingsEvent.value)
+
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_POSITIVE,
+            actionResId = R.string.actions_go_to_settings
+        )
+
+        assertNotNull(viewModel.onGoToSettingsEvent.value)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase3() {
+        //given
+        viewModel.onStart()
+        Mockito.clearInvocations(mockConnectionsRepository, mockApiManagerV1, mockApiManagerV2)
+        val isConnected = true
+        viewModel.onNetworkConnectionChanged(isConnected = isConnected)
+
+
+        //when
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_NEGATIVE,
+            actionResId = R.string.actions_retry,
+            guid = "guid1"
+        )
+
+        //then
+        assertNull(viewModel.onDeleteClickEvent.value)
+
+        //when
```

Comment:
each test case should not have multiple given-when-then annotation. split to different test cases